rules/consistent-function-scoping.js

```diff
@@ -90,18 +90,18 @@ const reactHooks = new Set([
 	'useLayoutEffect',
 	'useDebugValue'
 ]);
-const isReactHook = scope =>
-	scope.block &&
-	scope.block.parent &&
-	scope.block.parent.callee &&
-	scope.block.parent.callee.type === 'Identifier' &&
-	reactHooks.has(scope.block.parent.callee.name);
-
-const isArrowFunctionWithThis = scope =>
-	scope.type === 'function' &&
-	scope.block &&
-	scope.block.type === 'ArrowFunctionExpression' &&
-	(scope.thisFound || scope.childScopes.some(scope => isArrowFunctionWithThis(scope)));
+const isReactHook = ({block}) =>
```

Comment:
It's no longer possible to infer from the function what type it supports (that it supports a **scope**...